### PR TITLE
fix: none-mode auto-approve setup must fail open

### LINKED ISSUE #1978: [BUG] none-mode auto-approve setup fails closed - a discovery-layer failure tears down the unauthenticated webhook
Follow-up from my post-merge re-review of #1976, reviewed at `master` 4911d09d - every file cited below is unchanged since the merge (identical blobs at PR head `332b7ced`). @kingpanther13 asked me to re-review post-merge and flag anything for a follow-up PR on the next stable release.

## Main item: none-mode auto-approve setup fails closed

In `async_register_webhook`, the none-mode branch

```python
_register_metadata_views(hass)
bind_autoapprove_views(hass)
cfg[CFG_AUTOAPPROVE_PROVIDER] = AutoApproveProvider()
```

(`custom_components/ha_mcp_tools/mcp_webhook.py:738-740`) runs inside the `try` whose `except Exception:` at `:741` calls `async_unregister(...)`, closes the aiohttp session and re-raises - "Never leave a live endpoint (or a leaked session) behind a failed auth-setup path".

That is the right policy for `ha_auth` and `legacy`, where the failed piece *is* the authentication. In none mode it inverts the intent: the discovery documents plus the auto-approve authorization server are an enhancement layered on an intentionally unauthenticated webhook that otherwise always forwards, so any exception there takes down a webhook the user explicitly configured to need no auth.

The add-on flavor of the same feature already does the opposite and documents why - `homeassistant-addon-webhook-proxy-dev/mcp_proxy_dev/__init__.py:587` (`_setup_none_autoapprove`), docstring at `:599`:

> Fails OPEN, unlike ha_auth/legacy: the user did NOT opt into auth here (the webhook is intentionally unauthenticated), and this discovery is an enhancement layered on top of the already-working proxy. A failure must not tear the working webhook down [...]

So the two flavors of one fix diverge on exactly the property the fix exists to guarantee.

**Suggested fix:** wrap the component's none-mode branch in its own `try/except Exception: _LOGGER.exception(...); return`, mirroring `_setup_none_autoapprove`, and add a unit test that raises inside the none-mode discovery block and asserts the webhook stays registered. The existing `test_registration_failure_closes_session_and_unregisters` (`tests/src/unit/test_mcp_webhook.py:1139`) does not cover this: it injects the failure into `async_register` itself (`:1146`), where tearing down is correct because the webhook never bound.

**Fold in while there:** both view-binding helpers set their guard flag only after all `register_view` calls succeed - `bind_autoapprove_views` at `custom_components/ha_mcp_tools/oauth_autoapprove.py:301-303` (two views) and `_register_metadata_views` at `mcp_webhook.py:487-491` (seven views in a loop). If one raises partway, the already-bound views stay bound while the flag is unset, so a retry would re-register routes that are already claimed.

## Also in the same sweep

1. **Pending component version was never opened.** `custom_components/ha_mcp_tools/const.py:27` and `manifest.json:25` are `1.2.2` on master; `git show stable:custom_components/ha_mcp_tools/const.py` is also `1.2.2`. Per AGENTS.md ("Level with stable (no pending version yet): bump once - patch by default - to open the pending version"), #1976's component change should have opened `1.2.3`.

2. **`/authorize` is anonymous and can park the pending-code store.** It is at a fixed path and requires neither the webhook id nor a client identity, so anyone who knows the HA origin can submit valid S256 challenges with the public claude.ai callback until `MAX_PENDING_CODES` (1000, `custom_components/ha_mcp_tools/oauth_legacy.py:95`) is reached; legitimate onboarding then receives `temporarily_unavailable` until codes expire (`AUTH_CODE_TTL`, 5 minutes, `:71`). Self-healing, no data exposure - risk-accepting it is defensible, but the decision should be written down rather than implicit.

3. **Twin parity, error-response headers.** The add-on's `_json_error` (`homeassistant-addon-webhook-proxy-dev/mcp_proxy_dev/oauth_autoapprove.py:128-135`) omits the `Cache-Control: no-store` / `Pragma: no-cache` headers its component twin sets (`custom_components/ha_mcp_tools/oauth_autoapprove.py:104`). RFC 6749 scopes the MUST to responses carrying tokens or credentials, so this is parity rather than a spec violation - the RFC's own §5.2 example does carry the headers.

4. **Docstring vs behaviour, `PKCECodeStore.consume_code`.** `custom_components/ha_mcp_tools/oauth_legacy.py:438-439` claims the code is "popped (one-shot) before any check that can fail, so a failed attempt still burns it", but the two verifier-shape checks return before the `pop` at `:448`.

5. **Stale docstring plus duplicated literal in the add-on.** `_active_oauth_mode` (`homeassistant-addon-webhook-proxy-dev/mcp_proxy_dev/oauth.py:347`, docstring at `:353`) still documents `MODE_HA_AUTH` or `MODE_LEGACY` only, although it now also returns `none_autoapprove`; that literal is defined twice (`oauth.py:96` and `__init__.py:74`).

Items 1 and 2 are decisions rather than mechanical fixes; 3-5 are one-liners that fit alongside the main item.

### PR DESCRIPTION
**Pull request by kingpanther13**
## What does this PR do?

Post-merge follow-up to #1976 (issue #1978). In none mode the webhook is
intentionally unauthenticated, but the auto-approve discovery setup ran inside
the shared `try` whose `except` unregisters the webhook + closes the session —
so a failure in a cosmetic discovery layer tore down a webhook the user
configured to need no auth. The add-on twin already fails open
(`_setup_none_autoapprove`); this makes the component match.

- **Fail open (main):** wrap the none-mode discovery block in its own
  `try/except` that logs and continues; the webhook stays registered and
  forwarding. Provider is assigned last, so a partial bind leaves
  none-autoapprove inactive (plain proxy) rather than half-enabled. New unit
  test asserts the webhook survives a failure injected into the discovery block.
- **Guard-flag semantics:** `_register_metadata_views` / `bind_autoapprove_views`
  set their "bound once" flag only *after* the full bundle registers (per Codex
  review), so a partial bind stays distinguishable and none-autoapprove never
  advertises discovery with an unbound route. Regression test added.
- **Anonymous `/authorize` (concern 3):** documented as an accepted risk — the
  endpoint is anonymous by design (zero-login none mode); filling the pending-
  code store only delays a *first* connect's OAuth-discovery fallback by ≤5 min
  (self-healing), grants no access, and never affects the forwarding webhook.
- **`consume_code` docstring:** corrected to match the intentional ordering
  (shape guard runs before the one-shot pop).
- Opens the pending component version `1.2.3`.

The two add-on minors from #1978 (error-response header parity; a stale
`_active_oauth_mode` docstring + duplicated literal) ride the stable promotion
PR #1977, not this one.

## Type of change
- [x] 🐛 Bug fix

## Testing
- [x] All automated tests pass — 199 component unit tests green
- [x] Code follows style guidelines (`uv run ruff check`)

## Checklist
- [x] I have updated documentation if needed

Closes #1978

### CODE REVIEW

**Comment by ghhamcp** [comment]
@codex review — apply the review criteria in .gemini/styleguide.md in addition to AGENTS.md guidance

File: custom_components/ha_mcp_tools/oauth_autoapprove.py [hunk 1]
```diff
@@ -298,6 +310,10 @@ def bind_autoapprove_views(hass: HomeAssistant) -> None:
     """
     if hass.data.get(_AUTOAPPROVE_VIEWS_REGISTERED_KEY):
         return
+    # Claim the bind BEFORE registering (issue #1978): see
+    # mcp_webhook._register_metadata_views — one bind attempt per HA session so a
+    # partial failure can't wedge a retry into re-registering an already-bound
+    # view. Pairs with the none-mode caller's fail-open handling.
+    hass.data[_AUTOAPPROVE_VIEWS_REGISTERED_KEY] = True
     hass.http.register_view(AutoApproveAuthorizeView(hass))
     hass.http.register_view(AutoApproveTokenView(hass))
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply custom_components/ha_mcp_tools/oauth_autoapprove.py]: Fixed in `cf366e2c`. Reverted to setting the flag only **after** both views register, so it means the full bundle is bound. A partial bind now leaves the flag unset; the none-mode caller fails open (the retry's duplicate `register_view` is caught harmlessly) and never assigns a provider that advertises OAuth with an unbound `/token`. Added `test_partial_metadata_bind_leaves_guard_flag_unset` to lock the pattern.

File: custom_components/ha_mcp_tools/mcp_webhook.py [hunk 2]
```diff
@@ -486,9 +486,15 @@ def _register_metadata_views(hass: HomeAssistant) -> None:
...
         return
+    # Claim the bind BEFORE registering (issue #1978): aiohttp can't unregister a
+    # view and a duplicate register_view raises, so if this loop fails partway, a
+    # later retry (e.g. a mode switch) must NOT re-register the views that already
+    # bound. Setting the guard first caps this at one bind attempt per HA session
+    # — a partial failure leaves some discovery views unbound (they 404 like an
+    # unregistered route) instead of throwing a duplicate-route error next call.
+    hass.data[_OAUTH_VIEWS_REGISTERED_KEY] = True
     for view in _metadata_views(hass):
         hass.http.register_view(view)
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply custom_components/ha_mcp_tools/mcp_webhook.py]: Fixed in `cf366e2c`. Same change here: the flag is set only after all metadata views register, so a partial bind stays distinguishable and discovery is never advertised with an unbound metadata route. Covered by the regression test noted on the sibling `bind_autoapprove_views` thread.

File: custom_components/ha_mcp_tools/const.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 # manifest bump that forgets this constant (or vice-versa) fails in CI. The
 # capability negotiation — not this version — gates each WS command (see
 # ``websocket_api.CAPABILITIES``).
-COMPONENT_VERSION = "1.2.2"
+COMPONENT_VERSION = "1.2.3"
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply custom_components/ha_mcp_tools/const.py:27]: Already addressed: both pins were bumped to `1.2.3` — `test_component_ws_phase2_async.py:147` in `f57c8128` and `test_component_ws_search.py:771` in `29d64894`. Unit Tests is green on the latest run.

**Comment by ghhamcp** [comment]
@codex review — apply the review criteria in .gemini/styleguide.md in addition to AGENTS.md guidance

File: custom_components/ha_mcp_tools/mcp_webhook.py [hunk 4]
```diff
@@ -486,6 +486,13 @@ def _register_metadata_views(hass: HomeAssistant) -> None:
     """
     if hass.data.get(_OAUTH_VIEWS_REGISTERED_KEY):
         return
+    # Set the flag only AFTER every view registers (issue #1978): it must mean
+    # "the full bundle is bound", so a partial bind stays distinguishable from a
+    # complete one. Marking it bound early would let a later setup assign a
+    # provider and advertise discovery while some RFC metadata routes are still
+    # unbound — a 404 for the clients that probe them. On a partial bind the flag
+    # stays unset; the none-mode caller then fails open (the retry's duplicate
+    # register is caught harmlessly) while ha_auth/legacy fail closed.
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply custom_components/ha_mcp_tools/mcp_webhook.py:495]: Holding on `flag-last` here — I don't think this partial bind is reachable, so neither flag ordering is a real defect.

`register_view` binds our seven views all-or-nothing in practice. The URLs are component-namespaced (`{OAUTH_BASE}/…` and `/.well-known/…{OAUTH_BASE}`), so nothing external owns them to collide on view *N>1* while view 1 succeeds; and a frozen aiohttp router rejects the **first** view (nothing binds — a clean retry), with no `await` in the loop for the app to freeze mid-way. So there is no "some bound, one failed" state to strand.

This is the mirror of the earlier P2 on `flag-first` — both require the same mid-bundle partial bind. `flag-last` is the original shipped behavior and is kept in lockstep with the webhook-proxy add-on twin; it never advertises a partial bundle. The unit test guards the flag-integrity invariant defensively, not a reachable path.

Resolving as working-as-intended — happy to revisit with a concrete trigger where view *N>1* raises while views `1..N-1` stayed bound.
